chore(angularjs): Move AngularJS bindings to own package

asialgearoid opened the pull request:
As per discussion in #2476, this commit moves the AngularJS bindings into a new package
called angularjs-onsenui. For now, the bindings continue to be packaged with the core,
but with an added warning to users that they should take the opportunity to upgrade to
the new package before 2.11.0, when the bindings will be removed from the core package.

There are a large number of changed files, but the majority are test files that now reference the new location of `angularjs-onsenui.js`, as well as the new `bindings/angular1/node_modules` folder.

To go with this PR, I have also created a PR to update the docs with some info on how to migrate, which is linked from the warning message mentioned above. Though the link won't work yet, I'll make sure the docs are up before we publish 2.10.4.

Here are the test steps I have done:
* Run the e2e AngularJS tests
* Build the bindings using `gulp angular-bindings` from the project root, which is how it is currently done. I confirmed this builds the bindings to the `bindings/angular1/dist` folder, and also copies them to `build/js`. So, though the process is different, the result is the same as before, for now.
* Run `npm pack` in `bindings/angular1` to confirm what would be in the package if it were published. It contains `dist/angularjs-onsenui.js`, `dist/angularjs-onsenui.min.js`, `CHANGELOG.md` and `package.json`.
* Successfully built on Circle CI (all tests are passing)

emccorson commented:
I can't get this to build with the instructions in `CONTRIBUTING.md`:

```
cd css-components
yarn install
cd ..
yarn install
yarn build
```

`yarn build` output:

```
yarn run v1.7.0
warning ../../../package.json: No license field
$ gulp build
[18:54:26] Requiring external module babel-register
[18:54:33] Loading css-components/gulpfile.js
[18:54:36] Using gulpfile ~/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js
[18:54:36] Starting 'build'...
[18:54:36] Starting 'clean'...
[18:54:37] Finished 'clean' after 363 ms
[18:54:37] Starting 'core'...
┌─────────────────────────────────────────────────────┐
│                                                     │
│   Bundle size: 988.83 KB, Gzipped size: 190.68 KB   │
│                                                     │
└─────────────────────────────────────────────────────┘
[18:54:53] Finished 'core' after 17 s
[18:54:53] Starting 'core-esm'...
┌─────────────────────────────────────────────────┐
│                                                 │
│   Bundle size: 149 KB, Gzipped size: 36.17 KB   │
│                                                 │
└─────────────────────────────────────────────────┘
[18:54:58] Transpiling core ESM files ...
[18:55:02] Finished transpiling core ESM files
[18:55:02] Finished 'core-esm' after 8.9 s
[18:55:02] Starting 'angular-bindings'...
[18:55:03] 'angular-bindings' errored after 320 ms
[18:55:03] Error: ENOENT: no such file or directory, open 'bindings/angular1/dist/angularjs-onsenui.js'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:105:50
    at Array.forEach (<anonymous>)
    at copyAngularBindings (/Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:101:22)
    at Gulp.<anonymous> (/Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:112:3)
    at module.exports (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:134:8)
    at runNextSet (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/run-sequence/index.js:123:15)
    at Gulp.onTaskEnd (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/run-sequence/index.js:101:4)
    at emitOne (events.js:121:20)
    at Gulp.emit (events.js:211:7)
    at Gulp.Orchestrator._emitTaskDone (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:264:8)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:275:23
    at finish (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:21:8)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:43:4
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
[18:55:03] 'build' errored after 26 s
[18:55:03] Error in plugin 'run-sequence(angular-bindings)'
Message:
    ENOENT: no such file or directory, open 'bindings/angular1/dist/angularjs-onsenui.js'
Details:
    errno: -2
    code: ENOENT
    syscall: open
    path: bindings/angular1/dist/angularjs-onsenui.js
Stack:
Error: ENOENT: no such file or directory, open 'bindings/angular1/dist/angularjs-onsenui.js'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:105:50
    at Array.forEach (<anonymous>)
    at copyAngularBindings (/Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:101:22)
    at Gulp.<anonymous> (/Users/asial/src/now-is-it-good-news/OnsenUI/gulpfile.babel.js:112:3)
    at module.exports (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:134:8)
    at runNextSet (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/run-sequence/index.js:123:15)
    at Gulp.onTaskEnd (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/run-sequence/index.js:101:4)
    at emitOne (events.js:121:20)
    at Gulp.emit (events.js:211:7)
    at Gulp.Orchestrator._emitTaskDone (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:264:8)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/index.js:275:23
    at finish (/Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:21:8)
    at /Users/asial/src/now-is-it-good-news/OnsenUI/node_modules/orchestrator/lib/runTask.js:43:4
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

asialgearoid commented:
Good spot, @emccorson. The AngularJS bindings need to be built before building the core, or else nothing will be copied in. I'm glad that it fails properly!

I had updated the Circle CI script, but forgot the build instructions. I've changed them now. This will only be needed until 2.11.0, when we remove the bindings from the core, so we should update the instructions again when we do that.
